### PR TITLE
Add "NO_LOGGING" flavors for CONFIG_REQUIRED and CONFIG_OPTIONAL

### PR DESCRIPTION
**Pull request by dcristoloveanu**

### CODE REVIEW

File: tests/sf_service_config_ut/sf_service_config_ut.c [hunk 1]
```diff
@@ -90,7 +92,9 @@ TEST_SF_SERVICE_CONFIG_DEFINE_EXPECTED_CALL_HELPERS(my_config, expected_config_p
     CONFIG_OPTIONAL(thandle_rc_string, mocked_string_option_in_thandle_optional), \
     CONFIG_OPTIONAL(char_ptr, mocked_string_option_optional), \
     CONFIG_OPTIONAL(wchar_ptr, mocked_wide_string_option_optional), \
-    CONFIG_REQUIRED(bool, mocked_another_flag) \
+    CONFIG_REQUIRED(bool, mocked_another_flag), \
+    CONFIG_REQUIRED_NO_LOGGING(thandle_rc_string, a_secret), \
+    CONFIG_OPTIONAL_NO_LOGGING(char_ptr, another_secret) \
```
Comment (darobs) [inline tests/sf_service_config_ut/sf_service_config_ut.c:98]: this is the minimal set required for this PR. We really ought to have all types represented here.

**Review by darobs — APPROVED** [review]
:shipit:

File: tests/sf_service_config_ut/sf_service_config_ut.c (hunk 1 above)
Comment (dcristoloveanu) [reply tests/sf_service_config_ut/sf_service_config_ut.c:114]: Added entries for uint8_t, char_ptr, thandle_rcsrting, double and bool for both config and optional.

**Review by jasmineymlo — APPROVED** [review]

**Pull request merged**